Remove gauge card transparency on focus

karwosts opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->

## Proposed change

Gauge card currently turns transparent when focused. None of the other main card types exhibit this behavior. Remove this property from the css. It's not easy to notice against the default dark background, but when set against a different background it's pretty obvious. 

![gauge-transparent](https://user-images.githubusercontent.com/32912880/207697601-d8ea8f3b-f3fa-4d02-9220-79ea80fddb8c.gif)


## Type of change

<!--
  What type of change does your PR introduce to the Home Assistant frontend?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

<!--
  Supplying a configuration snippet, makes it easier for a maintainer to test
  your PR.
-->

```yaml

```

## Additional information

<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #12914
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

<!--
  Thank you for contributing <3
-->

[docs-repository]: https://github.com/home-assistant/home-assistant.io

karwosts committed:
Remove gauge card transparency on focus

karwosts committed:
Merge branch 'dev' of https://github.com/karwosts/frontend into gauge-12914

spacegaier commented:
Can you have a look at the button card as well, since I see the same CSS there in the coding => probably also should get removed.

karwosts committed:
Remove ha-card:focus from button card

karwosts commented:
I also removed the ha-card:focus from the button card, though that one is more subtle, since it does not modify the background color. All it does is set "outline: none", but the card don't seem to have an outline anyway, so I can't visually see that it does anything. 

Removed it just for consistency/cleanup.

bramkragten commented on src/panels/lovelace/cards/hui-gauge-card.ts:
```diff
@@ -268,11 +268,6 @@ class HuiGaugeCard extends LitElement implements LovelaceCard {
 
-      ha-card:focus {
-        outline: none;
-        background: var(--divider-color);
```
I agree this is the wrong color to use here, but we should still have a indicator the card is focussed. I suggest we use `background-color: var(--mdc-ripple-color, #000);` with an opacity of 0.12 as an `before`, the same as the button card has.

karwosts replied:
Are you suggesting theres something special about the gauge card that it needs a focus indicator? Other display type cards do not have this. Entity card, sensor card, glance card, graph card, weather card, etc.  don't have a focus indicator. Should they all be consistent, are you asking to add this to all those cards? 

And why opacity of 0.12, isn't that essentially close to fully transparent? (The thing I am trying to remove here?) Though I may not be understanding by what you mean "as an `before`". 

Entities card has focus on individual row icons only, not the text. Tile card has focus on icon and text separately as two elements. 

I'd need some guidance where to go from here, if you don't want to accept this change.

spacegaier replied:
I agree with @karwosts. The current behavior seems inconsistent. No other card can be focused visually, only elements on them.

Which makes me actually think we need the `outline: none` part since otherwise browser could potentially draw a focus outline (I just remembered this PR of mine https://github.com/home-assistant/frontend/pull/14535) and should only remove the background color statement.

bramkragten replied:
The gauge is an interactive element? You can click it and it will open the more info?

bramkragten replied:
Every element that is interactive and is focusable should have a focus indicator. That other cards are wrong does not make it right to remove it from this card.

steverep replied:
For accessibility, @bramkragten is definitely correct that interactive elements need a visual indicator that they are focused.  And that indicator needs to also have sufficient color contrast.

That said, this is a small part of a much larger problem with regard to more-info dialogs (and also user configurable tap actions).  Many cards or parts of cards are not even focusable via `tabindex`, let alone have a visual indicator.  They also do not have proper interactive ARIA roles.

I've been avoiding tackling this issue because it's quite complicated and cards are not consistent as you're finding out here.  The workaround for accessibility is just to learn that these things are clickable and not rely on tab to navigate them.

karwosts replied:
Bram if you don't want this current change, I have to admit I still don't understand your suggestion:

>  I suggest we use background-color: var(--mdc-ripple-color, #000); with an opacity of 0.12 as an before, the same as the button card has.

I looked at the button card it doesn't have anything like a background color, 

Either maybe you want to elaborate a bit, or if not I can just close the pr, but I would be sad leaving the gauge the way it currently is, that looks pretty terrible in the gif in the initial post.

bramkragten replied:
The button card uses the `mwc-ripple`

karwosts committed:
add back outline:none

karwosts commented:
I put back outline none, so now this only removes the background color on focus, which makes gauge card consisistent with all the other display-type cards (cards which are non-interactive, except to click it brings up "more info" for the target entity"). 

If there is desire to have all cards change color when focused, I think that might make sense as a new PR after more discussion with the UI team?

bramkragten commented:
Going to merge this, we should add some indication for focus in a follow up PR
